project/frontend/src/components/ButtonGroup.js

```diff
@@ -9,11 +9,38 @@ import React from 'react';
  */
 function ButtonGroup(props) {
   const { labelList } = props;
-  const buttonList = [];
-  for (let delta = 0; delta < labelList.length; delta++) {
-    buttonList.push(<button>{labelList[delta]}</button>);
-  }
-  return <div className='btn-group'>{buttonList}</div>;
+  const [backgroundIsGray, setBackgroundIsGray] = useState({
+    button0: false,
+    button1: false,
+    button2: false,
+    button3: false,
+  });
+
+  const onClickButton = (props) => {
+    let tempBackgroundIsGray = Object.assign({}, backgroundIsGray);
+    tempBackgroundIsGray[props] = !tempBackgroundIsGray[props];
+    setBackgroundIsGray(tempBackgroundIsGray);
+  };
+
+  const createButtons = () => {
+    const buttonList = [];
+    for (let delta = 0; delta < labelList.length; delta++) {
+      let className = backgroundIsGray[delta]
```

Comment:
Probably should check for index out of bounds in case labelList is longer that backgroundIsGray. And/or the initialization of backgroundIsGray could be based on however long labelList is?

Reply:
yup I was going to do this later today! I need to change the way that I initialize backGroundIsGray because it isn't dynamic